saleor/dashboard/menu/utils.py

```diff
@@ -92,3 +93,8 @@ def get_menus_that_needs_update(collection=None, categories=None, page=None):
     menus_to_be_updated = MenuItem.objects.filter(q).distinct().values_list(
         'menu', flat=True)
     return menus_to_be_updated
+
+
+def get_menu_obj_text(obj):
+    return str(obj) + ('' if obj.is_published else pgettext(
```

Comment:
So I have a doubt if that would work properly (Adding a translated string to nontranslated one), it might be not clear for the translators.

I'd say we need 3 cases here:
- Regular menu item
- Menu item available on a certain date (from the future)
- Menu item not published

and I'd add individual translation for 2nd and 3rd one

Reply:
You mean it's necessary to split the status "Not published" into 2 cases: 1) Not published collection or invisible page 2) Visible page since some data from the future?
Regarding concatenation of translated string with non-translated - it's possible to add an additional field "status" which will contain translated string to JSON response and join "text" and "status" on front-side, but I'm afraid it can complicate a little bit JS code - now there's common initialization for all selects. What would You advice?

Reply:
I’d just split it with if-else statement.

If not obj.is_published:
   Return pgettext(„Some description”, „%(page_name)s (not published)”  % str(page))
Elif obj.available_on > today:
   Return ...
Return str(obj)

Reply:
Ok, then I'll add additional status for menu items visible from a certain date in other places where `MenuItem` status appears as well